release/serve_tests/workloads/microbenchmarks.py

```diff
@@ -0,0 +1,185 @@
+"""Runs benchmarks.
+
+Latency benchmarks:
+    Runs a no-op workload with 1 replica.
+    Sends 100 requests to it and records average, P50, P90, P95, P99 latencies.
+
+Throughput benchmarks:
+    Asynchronously send batches of 100 requests.
+    Calculate the average throughput achieved on 10 batches of requests.
+"""
+import asyncio
+from functools import partial
+import json
+import logging
+
+import click
+import grpc
+import pandas as pd
+import requests
+from typing import Dict, List
+
+from ray import serve
+from ray.serve._private.benchmarks.common import (
+    Benchmarker,
+    do_single_grpc_batch,
+    do_single_http_batch,
+    Hello,
+    Noop,
+    run_latency_benchmark,
+    run_throughput_benchmark,
+)
+from ray.serve.generated import serve_pb2, serve_pb2_grpc
+from ray.serve.config import gRPCOptions
+from ray.serve.handle import DeploymentHandle
+
+from serve_test_utils import save_test_results
+
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+
+# For latency benchmarks
+NUM_REQUESTS = 100
+
+# For throughput benchmarks
+BATCH_SIZE = 100
+NUM_TRIALS = 10
+TRIAL_RUNTIME_S = 1
```

Comment:
would suggest runningmore trials & longer trial runtime (want to reduce noise as much as possible for release tests)

maybe 50 trials @ 5s each?